package-res/resources/web/pentaho/visual/base/View.js

```diff
@@ -180,22 +297,107 @@ define([
+     * [rejected:update]{@link pentaho.visual.base.events.RejectedUpdate} and
+     * the resolution of the returned promise.
+     *
+     * The [isUpdating]{@link pentaho.visual.base.View#isUpdating} property is `true` while the
+     * update operation is considered in progress,
+     * a period during which calling [update]{@link pentaho.visual.base.View#update} again
+     * returns the same promise that was returned by the initial `update` call.
+     *
+     * If the view is not currently updating and is also not [dirty]{@link pentaho.visual.base.View#isDirty},
+     * calling `update` _immediately_ returns a fulfilled promise.
+     *
+     * Otherwise, by this time, [isUpdating]{@link pentaho.visual.base.View#isUpdating} will be `true`,
+     * and the update operation proceeds to the **Will phase**.
+     *
+     * #### Will phase
+     *
+     * Initially, the [_updateWill]{@link pentaho.visual.base.View#_updateWill} method is called,
+     * which by default emits the [will:update]{@link pentaho.visual.base.events.WillUpdate} event.
+     * Either the implementation or the event listeners are allowed to cancel the update or further modify the model.
+     *
+     * If the update is _canceled_, the update is rejected with the cancel reason,
+     * and enters the **Rejected phase**.
      *
-     * Executes [_update]{@link pentaho.visual.base.View#_update} asynchronously in
-     * the will/did/rejected event loop associated with an update of a visualization,
-     * and also creates the visualization DOM node the first time it successfully updates.
+     * #### Loop phase
      *
-     * In order to get the visualization DOM node,
-     * listen to the {@link pentaho.visual.base.events.DidCreate|"did:create"} event.
+     * If the update is not canceled, the view enters an **update loop**,
+     * that only ends when either the view is up to date with the model,
+     * in which case surely [isDirty]{@link pentaho.visual.base.View#isDirty} will be `false`,
+     * or some error occurs.
      *
-     * @return {Promise} A promise that is fulfilled when the visualization
-     * is completely rendered. If the visualization is in an invalid state, the promise
-     * is immediately rejected.
+     * On each iteration of the update loop:
+     * 1. If the model is invalid, the update loop ends, with a validation error and
+     *    the update operation proceeds to the **Rejected phase**;
+     * 2. Otherwise, the "best fit" partial update method is selected and called to update the view;
+     * 3. If the selected update method throws an error or returns a rejected promise,
+     *    the update loop ends and the update operation proceeds to the **Rejected phase**;
+     * 4. If the view is not [dirty]{@link pentaho.visual.base.View#isDirty} anymore,
+     *    the update loop ends, with success;
+     * 5. Repeat.
+     *
+     * Over the view's lifetime, the very first "partial" update method that is selected is actually, always, the
```

Comment:
Over the view's lifetime, the very first "partial" update method that is selected is always the